src/fparser/two/Fortran2003.py

```diff
@@ -12030,40 +12038,58 @@ def match(string):
 
 
 class Stmt_Function_Stmt(StmtBase):  # R1238
-    """
-    <stmt-function-stmt>
-    = <function-name> ( [ <dummy-arg-name-list> ] ) = Scalar_Expr
+    """Implements the matching of a Fortran statement function statement.
+
+    stmt_function_stmt is
+                  function-name ( [ dummy-arg-name-list ] ) = scalar-expr
+
     """
 
     subclass_names = []
     use_names = ["Function_Name", "Dummy_Arg_Name_List", "Scalar_Expr"]
 
     @staticmethod
     def match(string):
-        i = string.find("=")
-        if i == -1:
-            return
-        expr = string[i + 1 :].lstrip()
+        """Implements the matching for an statement function statement.
+
+        :param str string: the string to match with as a statement
+            function statement.
+
+        :returns: a tuple of size 1 containing a Stmt_Function_Stmt \
+            object with the matched string if there is a match, or None \
+            if there is not.
+        :rtype: (:py:class:`fparser.two.Fortran2003.Stmt_Function_Stmt`) \
+            or NoneType
+
+        """
+        index = string.find("=")
+        if index == -1:
+            return None
+        expr = string[index + 1 :].lstrip()
         if not expr:
-            return
-        line = string[:i].rstrip()
+            return None
+        line = string[:index].rstrip()
         if not line or not line.endswith(")"):
-            return
-        i = line.find("(")
-        if i == -1:
-            return
-        name = line[:i].rstrip()
+            return None
+        index = line.find("(")
+        if index == -1:
+            return None
+        name = line[:index].rstrip()
         if not name:
-            return
-        args = line[i + 1 : -1].strip()
+            return None
+        args = line[index + 1 : -1].strip()
```

Comment:
At this point you could check whether the arguments are in the symbol table. If they are then this isn't a statement function (I think).

Reply:
stmt_function_stmt changes should not have been in this PR. I've removed them.